### PR TITLE
feat: make the minMetricSamplesToAlarm available for any alarm

### LINKED ISSUE #229: Feature suggestion: Add `minMetricSamplesToAlarm` to `ErrorRateThreshold`
I would like to suggest adding `minMetricSamplesToAlarm` to `ErrorRateThreshold` (or to its parent class `CustomAlarmThreshold`), so that one can add `minMetricSamplesToAlarm` to the `addFaultRateAlarm` entry in the props for `MonitoringFacade.monitorLambdaFunction()`.

There is no way I can find to specify either the requirement for statistical significance (`evaluateLowSampleCountPercentile: false` results in a CFN deployment failure saying that it is only applicable to percentile based metrics (which I had originally guessed that a rate would be, but it appears not) or a minimum number of samples when using
```
monitoringFacade.monitorLambdaFunction({
    ...
    addFaultRateAlarm: {
    },
    ...
});
```

### PR DESCRIPTION
**Pull request by voho**
Fixes #229 

---

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license_

### CODE REVIEW

File: lib/common/alarm/CustomAlarmThreshold.ts [hunk 1]
```diff
@@ -39,6 +39,18 @@ export interface CustomAlarmThreshold {
    */
   readonly alarmDescriptionOverride?: string;
 
+  /**
+   * Specifies how many samples (N) of the metric is needed to trigger the alarm.
+   * If this property is specified, an artificial composite alarm is created of the following:
```
Comment (echeung-amzn) [inline lib/common/alarm/CustomAlarmThreshold.ts:44]:
```suggestion
   * If this property is specified, an composite alarm is created of the following:
```

File: lib/common/alarm/CustomAlarmThreshold.ts [hunk 2]
```diff
@@ -39,6 +39,18 @@ export interface CustomAlarmThreshold {
...
+   * <ul>
+   * <li>The original alarm, created without this property being used; this alarm will have no actions set.</li>
+   * <li>A secondary alarm, which will monitor the same metric with the N (SampleCount) statistic, checking the sample count.</li>
+   * </ul>
+   * The newly created composite alarm will be returned as a result, and it will take the original alarm actions.
```
Comment (echeung-amzn) [inline lib/common/alarm/CustomAlarmThreshold.ts:49]:
```suggestion
   * This composite alarm will be returned as a result and uses the specified alarm actions.
```

**Review by echeung-amzn — COMMENTED** [review]
Minor wording suggestions.

File: lib/common/alarm/CustomAlarmThreshold.ts (hunk 1 above)
Comment (mavjop) [reply lib/common/alarm/CustomAlarmThreshold.ts:44]: If you remove "artificial", make sure to change the preceding "an" to an "a". :)

**Review by echeung-amzn — APPROVED** [review]

**Pull request merged**